File: client-api/src/test/java/org/ostelco/topup/api/auth/helpers/TestConfig.java
package org.ostelco.topup.api.auth.helpers;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.google.common.cache.CacheBuilderSpec;
import io.dropwizard.Configuration;
import io.dropwizard.client.JerseyClientConfiguration;
import javax.validation.Valid;
import javax.validation.constraints.NotNull;

public class TestConfig extends Configuration {
    @NotNull
    @Valid
    private String secret;

    @Valid
    @NotNull
    private CacheBuilderSpec cacheSpec;

    @Valid
    @NotNull
    private JerseyClientConfiguration jerseyClient = new JerseyClientConfiguration();

    @JsonProperty("secret")
    public void setSecret(final String secret) {
        this.secret = secret;
    }

    @JsonProperty("secret")
    public String getSecret() {
        return secret;
    }

    @JsonProperty("authenticationCachePolicy")
    public void setAuthenticationCachePolicy(final String spec) {
        this.cacheSpec = CacheBuilderSpec.parse(spec);
    }

    @JsonProperty("authenticationCachePolicy")
    public CacheBuilderSpec getAuthenticationCachePolicy() {
        return cacheSpec;
    }

    @JsonProperty("jerseyClient")
    public JerseyClientConfiguration getJerseyClientConfiguration() {
        return jerseyClient;
    }
}

File: client-api/src/test/kotlin/org/ostelco/prime/client/api/auth/helpers/TestConfig.kt
package org.ostelco.prime.client.api.auth.helpers

import com.fasterxml.jackson.annotation.JsonProperty
import com.google.common.cache.CacheBuilderSpec
import io.dropwizard.Configuration
import io.dropwizard.client.JerseyClientConfiguration
import javax.validation.Valid
import javax.validation.constraints.NotNull

class TestConfig : Configuration() {
    @NotNull
    @Valid
    @get:JsonProperty("secret")
    @set:JsonProperty("secret")
    var secret: String? = null

    @Valid
    @NotNull
    @get:JsonProperty("authenticationCachePolicy")
    var authenticationCachePolicy: CacheBuilderSpec? = null
        private set

    @Valid
    @NotNull
    @get:JsonProperty("jerseyClient")
    val jerseyClientConfiguration = JerseyClientConfiguration()

    @JsonProperty("authenticationCachePolicy")
    fun setAuthenticationCachePolicy(spec: String) {
        this.authenticationCachePolicy = CacheBuilderSpec.parse(spec)
    }
}
